### DISCUSSION TITLE
SOC web interface redirect

### DISCUSSION THREAD

**Discussion by kiaraho** [Security-Onion-Solutions/securityonion#1890] [Unsupported Versions] [answered]
I recently setup a new ManagerSearch node with two network interfaces in a test environment. The first interface (`10.0.2.13`) has external network access and the second (`10.48.0.10`) is internal only. During setup I configured the the second interface as the management interface and set SOC to use its IP address.

The output of `so-allow-view` includes `10.48.0.0/24`. However, when I try to access SOC from another system on this network, I get a browser redirect from `10.48.0.10` to `10.0.2.13`. Attempts to modify the IP in the URL results in an expected `CSRF token is missing or invalid` error. Here is the output of `/opt/so/conf/kibana/etc/kibana.yml`:

```
---
# Default Kibana configuration from kibana-docker.
server.name: kibana
server.host: "0"
server.basePath: /kibana
elasticsearch.hosts: [ "http://10.0.2.13:9200" ]
#kibana.index: ".kibana"
#elasticsearch.username: elastic
#elasticsearch.password: changeme
#xpack.monitoring.ui.container.elasticsearch.enabled: true
elasticsearch.requestTimeout: 90000
logging.dest: /var/log/kibana/kibana.log
telemetry.enabled: false
```

**Comment by shipler** [comment]
Try to login again from the same tab after rebrowsing the baseurl

**Reply by kiaraho** [reply]
Tried that several times. No luck.

Reran `so-setup-network` and reconfigured the server as a Standalone node this time. No more issues with URL redirects. Strange.

**Comment by kiaraho** [comment, marked as answer]
UPDATE
--------
As a test, I reran `so-setup-network` again and there were MySQL errors in `sosetup.log` similar to #1592. As a result of the install errors, some post-install scripts did not complete and the redirect issue reappeared, in addition to the Playbook issues in #1896. I reran `so-setup-network` three more times, including on a fresh Ubuntu 18 install and got the same results.

I had to manually change `url_base` to `10.48.0.10` in  `/opt/so/saltstack/local/pillar/global.sls`, despite picking this as the management interface during each `so-setup-network` run. Only then did the redirect issue get resolved. I guess it has something to do with the post-install scripts failing to run, though there is no clear reason why the install is failing now.
